booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,651 @@
+      allDg((dg) -> dg.path("imoClass").submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.DG_IMO_CLASSES)
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'segregationGroups' values must be from dataset",
+      allDg((dg) -> dg.path("segregationGroups").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.DG_SEGREGATION_GROUPS)
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The 'inhalationZone' values must be from dataset",
+      allDg((dg) -> dg.path("inhalationZone").all().submitPath()),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.INHALATION_ZONE_CODE)
+    ),
+    JsonAttribute.allOrNoneArePresent(
+      JsonPointer.compile("/declaredValue"),
+      JsonPointer.compile("/declaredValueCurrency")
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The charges 'currencyCode' values must be from dataset",
+      mav -> mav.submitAllMatching("charges.*.currencyCode"),
+      JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.ISO_4217_CURRENCY_CODES)
+    ),
+
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "The charges currency amount must not exceed more than 2 decimal points",
+      mav -> mav.submitAllMatching("charges.*"),
+      (nodeToValidate, contextPath) -> {
+        var currencyAmount = nodeToValidate.path("currencyAmount").asDouble();
+        if (BigDecimal.valueOf(currencyAmount).scale() > 2) {
+          return Set.of("currencyAmount %s is expected to have 2 decimal precious ".formatted(contextPath));
+        }
+        return Set.of();
+      }
+    )
+    );
+
+  private static final List<JsonContentCheck> RESPONSE_ONLY_CHECKS = Arrays.asList(
+    ADVANCED_MANIFEST_FILING_CODES_UNIQUE,
+    AMF_CC_MTC_COMBINATION_VALIDATIONS,
+    SHIPMENT_CUTOFF_TIMES_UNIQUE,
+    CHECK_CONFIRMED_BOOKING_FIELDS,
+    VALIDATE_SHIPMENT_LOCATIONS,
+    REQUESTED_CHANGES_PRESENCE,
+    REASON_PRESENCE
+  );
+
+  public static ActionCheck responseContentChecks(UUID matched, Supplier<CarrierScenarioParameters> cspSupplier, Supplier<DynamicScenarioParameters> dspSupplier, BookingState bookingStatus, BookingState amendedBookingState) {
+    var checks = new ArrayList<JsonContentCheck>();
+    checks.add(JsonAttribute.mustEqual(
+      CARRIER_BOOKING_REQUEST_REFERENCE,
+      () -> dspSupplier.get().carrierBookingRequestReference()
+    ));
+    checks.add(JsonAttribute.mustEqual(
+      BOOKING_STATUS,
+      bookingStatus.wireName()
+    ));
+    checks.addAll(STATIC_BOOKING_CHECKS);
+    checks.addAll(RESPONSE_ONLY_CHECKS);
+    if (CONFIRMED_BOOKING_STATES.contains(bookingStatus)) {
+      checks.add(COMMODITIES_SUBREFERENCE_UNIQUE);
+    }
+    if (BOOKING_STATES_WHERE_CBR_IS_OPTIONAL.contains(bookingStatus)) {
+      checks.add(JsonAttribute.mustBeAbsent(
+        CARRIER_BOOKING_REFERENCE
+      ));
+    }
```

Comment:
This check is not correct. The carrier booking reference can be present in any state.  However, it **must** be present in some states.

Also, we should validate that the carrier booking reference is the same once it appears (just as we do with the carrier booking _request_ reference).

Reply:
This `mustBeAbsent` check is still here, but as stated it is incorrect. The standard never forbids the CBR in any state. The standard requires the CBR in certain states.